core/src/main/java/org/mule/runtime/core/source/scheduler/DefaultSchedulerMessageSource.java

```diff
@@ -143,7 +144,8 @@ private void pollWith(final Message request) {
 
         @Override
         public Event process() throws Exception {
-          Event event = builder(create(flowConstruct, getPollingUniqueName())).message(request).flow(flowConstruct).build();
+          Event event = builder(create(flowConstruct, fromSingleComponent(getPollingUniqueName()))).message(request)
```

Comment:
maybe it was wrong before, but isn't `DefaultSchedulerMessageSource` a source with a `ComponentLocation`?